h5py/tests/test_file.py

```diff
@@ -780,3 +780,16 @@ def test_swmr_mode_consistency(self):
         # This setter should affect both fid and group member file attribute
         assert fid.swmr_mode == g.file.swmr_mode == True
         fid.close()
+
+# unittest doesn't work with pytest fixtures (and possibly other features),
+# hence no subclassing TestCase
+class TestROS3(object):
+    @pytest.mark.skipif(h5py.version.hdf5_version_tuple < (1, 10, 6)
+                        or not h5py.h5.H5PYConfig().ros3,
+                        reason="ros3 file operations were added in HDF5 1.10.6+")
+    def test_ros3(self):
+        """ ROS3 driver and options """
+
+        with File("https://s3.us-east-2.amazonaws.com/hdf5ros3/GMODO-SVM01.h5", 'r', driver='ros3') as f:
```

Comment:
AFAIK it was provided by hdf5 group as an example url and is public. We could of cause check with them or just ask for forgiveness later: I don't know if they are paying for data egress.

Reply:
Oh, cool, I'll post the question on the forum then (I was more concerned that it was one of @satra's files, and either it would produce a surprise Amazon bill or be taken down at some point).

Reply:
See https://forum.hdfgroup.org/t/test-url-for-ros3-driver/7986

Reply:
this has been fixed by shifting to one of our files.